packages/zowe-explorer/src/trees/shared/SharedInit.ts

```diff
@@ -338,40 +318,36 @@ export class SharedInit {
-            });
-            watcher.onDidChange(async (uri: vscode.Uri) => {
-                ZoweLogger.info(vscode.l10n.t("Team config file updated."));
-                const newProfileContents = Buffer.from(await vscode.workspace.fs.readFile(uri));
-                if (Constants.SAVED_PROFILE_CONTENTS.get(uri.fsPath)?.equals(newProfileContents)) {
-                    return;
-                }
-                Constants.SAVED_PROFILE_CONTENTS.set(uri.fsPath, newProfileContents);
-                void SharedActions.refreshAll();
-                ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.UPDATE);
-            });
+            watcher.onDidCreate(
+                SharedUtils.debounce(() => {
+                    ZoweLogger.info(vscode.l10n.t("Team config file created, refreshing Zowe Explorer."));
+                    void SharedActions.refreshAll();
+                    ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.CREATE);
+                }, 100) // eslint-disable-line no-magic-numbers
+            );
+            watcher.onDidDelete(
+                SharedUtils.debounce(() => {
+                    ZoweLogger.info(vscode.l10n.t("Team config file deleted, refreshing Zowe Explorer."));
+                    void SharedActions.refreshAll();
+                    ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.DELETE);
+                }, 100) // eslint-disable-line no-magic-numbers
+            );
+            watcher.onDidChange(
+                SharedUtils.debounce(() => {
+                    ZoweLogger.info(vscode.l10n.t("Team config file updated, refreshing Zowe Explorer."));
+                    void SharedActions.refreshAll();
+                    ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.UPDATE);
+                }, 100) // eslint-disable-line no-magic-numbers
+            );
         });
 
         try {
-            // Workaround to skip ON_VAULT_CHANGED events triggered by ZE and not by external app
-            // TODO: Remove this hack once https://github.com/zowe/zowe-cli/issues/2279 is implemented
-            SharedInit.originalEmitZoweEvent = (imperative.EventProcessor.prototype as any).emitZoweEvent;
-            (imperative.EventProcessor.prototype as any).emitZoweEvent = function (eventName: string): void {
-                SharedInit.emitZoweEventHook(this, eventName);
-            };
-
-            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(
-                imperative.ZoweUserEvents.ON_VAULT_CHANGED,
-                SharedInit.onVaultChanged
-            );
+            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(imperative.ZoweUserEvents.ON_VAULT_CHANGED, async () => {
+                ZoweLogger.info(vscode.l10n.t("Changes in the credential vault detected, refreshing Zowe Explorer."));
```

Comment:
I'm probably missing something, but unless I write my own nodejs script (that calls the `save` function from the config.api.secure properties) I don't see how this event is getting triggered 🤯

And even with a custom script, this event is not captured in debug mode (it is only captured in a VSIX installation) 😕 

Because of this, running multiple instances of this PR may cause credentials to not be updated properly between multiple windows of Zowe Explorer 😢